src/ports/squids/component.ts

```diff
@@ -0,0 +1,172 @@
+
+        if (taskArns.length === 0) continue;
+
+        const describeTasksCommand = new DescribeTasksCommand({
+          cluster: CLUSTER_NAME,
+          tasks: taskArns,
+        });
+        const describeResponse = await client.send(describeTasksCommand);
+        const tasks = describeResponse.tasks || [];
+
+        const squid: Partial<Squid> = {
+          name: squidService.serviceName || "",
+          service_name: squidService.serviceName || "",
+          metrics: {} as Record<Network.ETHEREUM | Network.MATIC, SquidMetric>,
+        };
+
+        // there should be just one task per service
+        for (const task of tasks) {
+          squid.version = task.version || 0;
+          squid.created_at = task.createdAt;
+          squid.health_status = task.healthStatus;
+          squid.service_status = task.lastStatus;
+
+          const ip = task.attachments
+            ?.find((att) => att.type === "ElasticNetworkInterface")
+            ?.details?.find(
+              (detail) => detail.name === "privateIPv4Address"
+            )?.value;
+
+          if (!ip) continue;
+
+          // Step 3: Fetch /metrics from each IP for each network
+          try {
+            for (const network of getSquidsNetworksMapping()) {
+              const response = await fetch.fetch(
+                `http://${ip}:${network.port}/metrics`
+              );
+              const text = await response.text(); // Use text() since the response is plain text
+
+              if (!squid.metrics) {
+                squid.metrics = {} as Record<
+                  Network.ETHEREUM | Network.MATIC,
+                  SquidMetric
+                >;
+              }
+
+              squid.metrics[network.name] = {
+                sqd_processor_sync_eta_seconds: getMetricValue(
+                  text,
+                  "sqd_processor_sync_eta_seconds"
+                ),
+                sqd_processor_mapping_blocks_per_second: getMetricValue(
+                  text,
+                  "sqd_processor_mapping_blocks_per_second"
+                ),
+                sqd_processor_last_block: getMetricValue(
+                  text,
+                  "sqd_processor_last_block"
+                ),
+                sqd_processor_chain_height: getMetricValue(
+                  text,
+                  "sqd_processor_chain_height"
+                ),
+              };
+            }
+          } catch (error) {
+            console.error(`Failed to fetch metrics for ${ip}:`, error);
+          }
+          if (squid.created_at && squid.health_status && squid.service_status) {
+            results.push(squid as Squid);
+          } else {
+            console.warn(`Skipping incomplete squid: ${squid.service_name}`);
+          }
+        }
+      }
+
+      return results;
+    } catch (error) {
+      console.error("Error listing services:", error);
+      return [];
+    }
+  }
+
+  async function stop(serviceName: string): Promise<void> {
```

Comment:
If I'm not wrong, by exporting the stop function, [the lifecycle mechanism of the components will run it upon termination](https://github.com/well-known-components/interfaces/blob/73d537fa11e1283d3df414779b7cc9f09b9c14c4/src/utils/lifecycle.ts#L3). Would you mind checking if we need to change the name of the funcion?

Reply:
good call, renamed!